agent/cache/cache.go

```diff
@@ -290,10 +291,10 @@ RETRY_GET:
 
 	// Get the current value
 	c.entriesLock.RLock()
-	_, cacheHit, entry := c.getEntryLocked(tEntry, key, info.MaxAge, info.MustRevalidate && first, minIndex)
+	_, entryValid, entry := c.getEntryLocked(tEntry, key, info.MaxAge, info.MustRevalidate && first, minIndex)
```

Comment:
Oh I think I see, you changed the return value on line 229 to `return ok, entry.Valid, entry` instead of returning `		return ok, cacheHit, entry`. Is that a valid change? 🤔

Reply:
I believe so, the return value should not have changed. I think `cacheHit` was a misleading name. It depends on what you consider a hit. I might argue the first return value is actually the cache hit, the second one says if the hit is valid or not.

I should change the second return value to be an explicit `false`. Right now it will always be false, because if `!ok` the `entry` will always be the zero value, so Valid is false, and the second condition explicitly checks it is false. I think the explicit `false` will be more obvious.

Reply:
I was hoping to remove one of the booleans in the return value, but one of the caller does handle the cases different. It's rather long function, so I haven't fully digested why it needs to care about these two cases differently.

Reply:
Seems reasonable to me. I would probably rename the variables in `fetch` as well so that the 2 places we use `getEntryLocked` match up.

Reply:
Good call, I've renamed the variable in `fetch` as well